aicsimageio/readers/czi_reader.py

```diff
@@ -0,0 +1,784 @@
+                consistent=czi.shape_is_consistent,
+            )
+
+            # Create the fill array shape
+            # Drop the YX as we will be pulling the individual YX planes
+            retrieve_shape: List[int] = []
+            use_selected_or_np_map: Dict[str, Optional[int]] = {}
+            for dim, index_op in zip(retrieve_dims, retrieve_indices):
+                if (
+                    dim not in [DimensionNames.SpatialY, DimensionNames.SpatialX, "A"]
+                    and dim in dims_shape.keys()
+                ):
+                    # Handle slices
+                    if index_op is None:
+                        # Store the dim for later to inform to use the np index
+                        use_selected_or_np_map[dim] = None
+                        if dim == DimensionNames.MosaicTile:
+                            retrieve_shape.append(
+                                dims_shape[DimensionNames.MosaicTile][1]
+                            )
+                        elif dim == DimensionNames.Time:
+                            retrieve_shape.append(dims_shape[DimensionNames.Time][1])
+                        elif dim == DimensionNames.Channel:
+                            retrieve_shape.append(dims_shape[DimensionNames.Channel][1])
+                        elif dim == DimensionNames.SpatialZ:
+                            retrieve_shape.append(
+                                dims_shape[DimensionNames.SpatialZ][1]
+                            )
+
+                    # Handle non-chunk dimensions (specific indices / ints)
+                    else:
+                        # Store the dim for later to inform to use the provided index
+                        use_selected_or_np_map[dim] = index_op
+                        retrieve_shape.append(1)
+
+            # Create list of planes that we will add each plane to, later we reshape
+            # Create empty arr with the desired shape to enumerate over the np index
+            planes: List[np.ndarray] = []
+            np_array_for_indices = np.empty(tuple(retrieve_shape), dtype=object)
+            for np_index, _ in np.ndenumerate(np_array_for_indices):
+                # Get each plane's index selection operations
+                # If the dimension is None, use the enumerated np index
+                # If the dimension is not None, use the provided value
+                plane_indices: Dict[str, int] = {"S": scene}
+
+                # Handle MosaicTile
+                if DimensionNames.MosaicTile in use_selected_or_np_map:
+                    if use_selected_or_np_map[DimensionNames.MosaicTile] is None:
+                        plane_indices["M"] = np_index[
+                            retrieve_dims.index(DimensionNames.MosaicTile)
+                        ]
+                    else:
+                        plane_indices["M"] = use_selected_or_np_map[  # type: ignore
+                            DimensionNames.MosaicTile
+                        ]
```

Comment:
For all of these checks, I think it might be a little more clear to structure this code like:
```suggestion
                if DimensionNames.MosaicTile in use_selected_or_np_map:
                    indices = use_selected_or_np_map[DimensionNames.MosaicTile]
                    if indices is None:
                        indices = np_index[
                            retrieve_dims.index(DimensionNames.MosaicTile)
                        ]
                    plane_indices["M"] = indices
```

Does this also need an `if DimensionNames.MosaicTile in retrieve_dims` check as well?